qiskit_experiments/data_processing/nodes.py

```diff
@@ -0,0 +1,244 @@
+
+        Raises:
+            DataProcessorError: if the data has no memory.
+        """
+        data[self.node_output] = self.kernel.kernel(np.array(data["memory"]))
+
+
+@discriminator
+@prev_node(Kernel)
+class Discriminator(DataAction):
+    """Backend system discriminator."""
+
+    def __init__(self, discriminator_, name: Optional[str] = None):
+        """
+        Args:
+            discriminator_: The discriminator used to transform the data to counts.
+                For example, transform IQ data to counts.
+            name: Optional name for the node.
+        """
+        self.discriminator = discriminator_
+        self.name = name
+        super().__init__()
+
+    @property
+    def node_output(self) -> str:
+        """Key under which Discriminator stores the data."""
+        return "counts"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["memory"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Discriminate the data to transform it into counts.
+
+        Args:
+            data: The data in a format that can be understood by the discriminator.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+        data[self.node_output] = self.discriminator.discriminate(np.array(data["memory"]))
+
+
+@iq_data
+@prev_node(Kernel)
+class ToReal(DataAction):
+    """IQ data post-processing. This returns real part of IQ data."""
+
+    def __init__(self, scale: Optional[float] = 1.0, average: bool = False):
+        """
+        Args:
+            scale: scale by which to multiply the real part of the data.
+            average: if True the single-shots are averaged.
+        """
+        self.scale = scale
+        self.average = average
+        super().__init__()
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToReal stores the data."""
+        return "memory_real"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["memory"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Modifies the data inplace by taking the real part of the memory and
+        scaling it by the given factor.
+
+        Args:
+            data: The data dict. IQ data is stored under memory.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+
+        # Single shot data
+        if isinstance(data["memory"][0][0], list):
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * _[0] for _ in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * _[0] for _ in data["memory"]]
```

Comment:
Maybe also return the standard deviation

Reply:
This is not doing any averaging so there is no standard dev to return here.